modules/gapi/src/backends/fluid/gfluidcore_func.simd.hpp

```diff
@@ -1536,6 +1583,355 @@ MULC_SIMD(float, float)
+            v_float32 a3 = vg_load_f32(&in[x + 2*nlanes]);
+
+            vx_store(&out[x], div_op(s_tag, a1, s1, v_scale));
+            vx_store(&out[x + nlanes], div_op(s_tag, a2, s2, v_scale));
+            vx_store(&out[x + 2*nlanes], div_op(s_tag, a3, s3, v_scale));
+        }
+
+        if (x < length)
+        {
+            x = length - lanes;
+            continue;  // process unaligned tail
+        }
+        break;
+    }
+    return x;
+}
+
+//-------------------------------------------------------------------------------------------------
+
+template<typename scale_tag_t, typename SRC, typename DST>
+CV_ALWAYS_INLINE int divc_mask_simd_c3(scale_tag_t s_tag, const SRC in[],
+                                       const float scalar[], DST out[],
+                                       const int length, const float scale)
+{
+    constexpr int chan = 3;
+    constexpr int nlanes = vector_type_of_t<DST>::nlanes;
+    constexpr int lanes = chan * nlanes;
+
+    if (length < lanes)
+        return 0;
+
+    v_float32 v_scale = vx_setall_f32(scale);
+
+    v_float32 s1 = vx_load(scalar);
+#if CV_SIMD_WIDTH == 32
+    v_float32 s2 = vx_load(&scalar[2]);
+    v_float32 s3 = vx_load(&scalar[1]);
+#else
+    v_float32 s2 = vx_load(&scalar[1]);
+    v_float32 s3 = vx_load(&scalar[2]);
+#endif
+     return divc_simd_c3_impl(s_tag, in, out, s1, s2, s3, v_scale, length, nlanes, lanes);
+}
+
+//-------------------------------------------------------------------------------------------------
+
+#define DIVC_SIMD(SRC, DST)                                                    \
+int divc_simd(const SRC in[], const float scalar[], DST out[],                 \
+              const int length, const int chan, const float scale,             \
+              const int set_mask_flag)                                         \
+{                                                                              \
+    switch (chan)                                                              \
+    {                                                                          \
+    case 1:                                                                    \
+    case 2:                                                                    \
+    case 4:                                                                    \
+    {                                                                          \
+        if (std::fabs(scale - 1.0f) <= FLT_EPSILON)                            \
+        {                                                                      \
+            if (set_mask_flag == 1)                                            \
+                return divc_mask_simd_common(not_scale_tag{}, in, scalar,      \
+                                             out, length, scale);              \
+            else                                                               \
+                return arithmOpScalar_simd_common(div_tag{}, in, scalar,       \
```

Comment:
What is difference `divc_mask_simd_common(not scale)` and `arithmOpScalar_simd_common(div)` ?

Reply:
> What is difference `divc_mask_simd_common(not scale)` and `arithmOpScalar_simd_common(div)` ?

Difference is in the presence or absence of a mask which must be calculated and applied to division result vector. It's necessary to find results of division by zero and replace them with zeros.

Reply:
I was confused with different names *divc_mask_xx* and *arithmOpScalar_xx(div)* for single  div operation. When reading code and compare div_ and arthm_  it is expected that operations would be differ but all of them is div with different meaning, which confused.
If the difference of both operation is just `scale` param presence then is it possible to use simple **function overloading** by params ? something like this

```
division(...., param)
division(...., param, scale)
```
?

Reply:
The `arithmOpScalar_simd_...() ` are used by all arithmetic operation with scalar. The `divc_mask_simd_...()` are used by DivC kernel only since it is the kernel specific functional.

Reply:
it would great to place this comment in **code** but not in review discussion, appreciate if you can do that :)